base/uk.ac.stfc.isis.ibex.ui.journalviewer/src/uk/ac/stfc/isis/ibex/ui/journalviewer/SearchInput.java

```diff
@@ -0,0 +1,309 @@
+            }
+        });
+
+        chkNumberTo.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                spinnerToNumber.setEnabled(chkNumberTo.getSelection());
+            }
+        });
+
+        cmbFilterType.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                DISPLAY.asyncExec(new Runnable() {
+                    @Override
+                    public void run() {
+                        stackSearch.topControl = cmpFilters.get(cmbFilterType.getSelectionIndex());
+                        cmpSearch.layout();
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * @return The current selection index of the filter type drop-down menu.
+     */
+    public int getCmbFilterTypeIndex() {
+        return cmbFilterType.getSelectionIndex();
+    }
+
+    /**
+     * @return The text in the search box of the currently active filter type.
+     */
+    public Optional<String> getActiveSearchText() {
+        if (stackSearch.topControl == cmpTitle) {
+            return Optional.of(txtSearchTitle.getText());
+        } else if (stackSearch.topControl == cmpRBNumber) {
+            return Optional.of(txtSearchRBNumber.getText());
+        } else if (stackSearch.topControl == cmpUsers) {
+            return Optional.of(txtSearchUsers.getText());
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * @return The number in the from run number spinner.
+     */
+    public Optional<Integer> getRunNumberFrom() {
+        return chkNumberFrom.getSelection() ? Optional.of(spinnerFromNumber.getSelection()) : Optional.empty();
+    }
+
+    /**
+     * @return The number in the to run number spinner.
+     */
+    public Optional<Integer> getRunNumberTo() {
+        return chkNumberTo.getSelection() ? Optional.of(spinnerToNumber.getSelection()) : Optional.empty();
+    }
+
+    /**
+     * @return A Calendar representing the start time to search from.
+     */
+    public Optional<Calendar> getStartTimeFrom() {
+        return chkTimeFrom.getSelection() ? Optional.of(new GregorianCalendar(dtFromDate.getYear(), dtFromDate.getMonth(),
+                dtFromDate.getDay(), dtFromTime.getHours(), dtFromTime.getMinutes(), dtFromTime.getSeconds())) : Optional.empty();
+    }
+
+    /**
+     * @return A Calendar representing the start time to search to.
+     */
+    public Optional<Calendar> getStartTimeTo() {
+        return chkTimeTo.getSelection() ? Optional.of(new GregorianCalendar(dtToDate.getYear(), dtToDate.getMonth(),
+                dtToDate.getDay(), dtToTime.getHours(), dtToTime.getMinutes(), dtToTime.getSeconds())) : Optional.empty();
+    }
+    
+    /**
+     * Clears all of the user input and resets the buttons.
+     */
+    public void clearInput() {
```

Comment:
This logic doesn't belong in the UI layer.

What we want to do is:
- Databind all of the UI items to a viewmodel
- The `clear` operation is implemented in the viewmodel, which updates things back to their defaults
- The databinding then takes care of updating the UI.

Reply:
After discussion we decided not to do this; the code is still maintainable the way it is without using bindings.